### ISSUE TITLE
[Flux IPadapter] Support Xlabs IPadapter in diffusers

### ISSUE THREAD

**Issue by chuck-ma** [huggingface/diffusers#9403]
It'd be great to have XLabs IPadapter supported in diffusers. 

Code: https://github.com/XLabs-AI/x-flux/
Checkpoint: https://huggingface.co/XLabs-AI/flux-ip-adapter/tree/main

@sayakpaul

**Comment by sayakpaul** [comment]
Feel free to start a PR if you're interested to contribute it. We can provide guidance along the way. 

Cc: @fabiorigano too

**Comment by chuck-ma** [comment]
Ok, I will have a try as long as I have time. Besides, I'm also curious if it will also work to make style transfer like we did in SDXL ipadapter.

**Comment by CONSTANT1386** [comment]
I have completed the training script and code for FLUX IP-Adapter based on diffusers, but I still encounter some issues in the final effect, Anyone can help me?

**Comment by asomoza** [comment]
Hi, you probably will have better luck asking in the original code repository, we don't even have official support for inference here yet.

**Comment by TinyTigerPan** [comment]
> I have completed the training script and code for FLUX IP-Adapter based on diffusers, but I still encounter some issues in the final effect, Anyone can help me?

Hi, I'm working on FLUX IP-Adapter training script based on x-flux. And I also meet some problem, maybe we can communicate together. Here is my email zhouxianpan@vip.qq.com

**Comment by max-fofanov** [comment]
Hello guys, I'm also interested in collaborating on this issue! My email is max.fofanov@gmail.com.

**Comment by vladmandic** [comment]
with popularity of flux, this is becoming one of the top user requests - and hints on possible implementation?

**Comment by sayakpaul** [comment]
Cc: @asomoza

**Comment by tariq659891** [comment]
Hello @sayakpaul,

I came across our GitHub repo and had the chance to read your Word doc—amazing work! I'm a big fan! While exploring the repo, I looked for areas where I could contribute and stumbled upon this one. I started working on it last night.

So far, I haven’t found any existing implementation of the IP adapter in Diffusers, so I decided to give it a shot. I’ve implemented it, and while it’s not producing high-quality results yet, I can see the impact of the IP adapter.

I’m sharing the code here for anyone who’s interested in checking it out or continuing the work.

Additionally, I’ve included inference code that runs on less than 17GB of memory, with plans to optimize it further to go below 14GB.

https://github.com/tariq659891/Fluxforge-inference/tree/main
